app/src/main/java/org/kiwix/kiwixmobile/main/AddNoteDialog.java

```diff
@@ -149,6 +158,60 @@ public void afterTextChanged(Editable s) {}
     return view;
   }
 
+  private String getZimNoteDirectoryName() {
+    String zimFileName = ZimContentProvider.getZimFile(); // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
+
+    String noteDirectoryName = getTextAfterLastSlashWithoutExtension(zimFileName);
+
+    if(noteDirectoryName != null) {
```

Comment:
Always be wary of
```
if (condition)
   return something
else
  return somethingElse
```
Can always be shortened into a ternary.
Seeing as how you are up to date with develop do you want to take a stab at converting this file to kotlin?
this return would get a lot more beautiful be using the elvis operator
```
return noteDiretoryName ?: zimFileTitle
```

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1237/commits/085ed7f7bd8c0916f1d1a3ad25a705a3bf1bdfd4#diff-36e13fa4a9132616aee81e1f88bfa0ffR166)

> Seeing as how you are up to date with develop do you want to take a stab at converting this file to kotlin?

Problem is I barely know any kotlin (Since I wasn't using it, didn't learn anything beyond the 4th excercise in kt playground). Is it possible to shift to it from java in a couple of weeks or so? Then I am willing to convert it once I'm done with the remaining gsoc deliverables

PS: Googled why `?:` is called the Elvis op, wasn't disappointed :laughing:

Reply:
Sounds good to me. It is  very well named operator